snappy/DiskCache_DiskCache.py

```diff
@@ -532,6 +533,20 @@ def evict(self, key):
         except:
             logger.log(logLevel.ERROR, "Unable to delete file evicted from cache: {}"
                        .format(toEvict.path))
+        self.removeEmptyCacheDirs(os.path.dirname(toEvict.path))
+
+    def removeEmptyCacheDirs(self, directory):
+        while directory != config['cachePath']:
+            # Going for the "easier to ask for forgiveness than permission" model:
```

Comment:
You want to go with this model in this case so as to avoid the typical race condition between check-and-delete. I mention that only because this comment might suggest you're saying you're taking a lazy route, where I think it's actually the correct route to take.

You don't have to change the comment or anything.